Fix some errors on shutdown

Damon3000s opened the pull request:
One error on shutdown that we are running into frequently is when git threads are running in the baackground and they attempt to access data from the GitModule. This access will attempt to get the module but if it isn't loaded it will try to load it. This causes an error because we don't allow the module to be loaded off the main thread. It isn't loaded because it has been unloaded due to editor shutdown.

Origin has "fixed" this by adding a bunch of thread safe get accessors which will get only and no load. If we returned no module then stop doing whatever operation you are doing. I have propagated that pattern into functions that reproduced the crash locally.

More details also exist in this ticket: https://thethreethousands.atlassian.net/browse/TECH-142

Damon3000s committed:
GitSourceControlModule thread safety  fix in RefreshLocks

Damon3000s committed:
Fix another thread safety crash

Damon3000s committed:
Merge branch 'dev-3k' into thread-safety-fixes

Damon3000s committed:
Merge branch 'dev-3k' into thread-safety-fixes

Darcy3000s commented on Source/GitSourceControl/Private/GitSourceControlUtils.cpp:
```diff
@@ -1494,7 +1500,7 @@ bool RefreshLocks(const FString& InRepositoryRoot, const FString& GitBinaryFallb
 		}
 
-		const FString& LfsUserName = FGitSourceControlModule::Get().GetProvider().GetLockUser();
+		const FString& LfsUserName = GitSourceControlModule->GetProvider().GetLockUser();
```
"git lfs locks" is a long operation - you're better off hoisting this to immediately after the safeguard above, it could be 5 seconds before we reach this point

Damon3000s replied:
Done.

Owen3000s reviewed (commented):
Looks good to me

Darcy3000s commented on Source/GitSourceControl/Private/GitSourceControlUtils.cpp:
```diff
@@ -1633,7 +1645,7 @@ bool RunUpdateStatus(const FString& InPathToGitBinary, const FString& InReposito
 	// NB:	It is important to still update the local status, as saving files doesn't mark them as modified in source control
 	//		The engine relies on this status update to update the file to reflect that it is modified
-	FGitSourceControlProvider& provider = FGitSourceControlModule::Get().GetProvider();
+	FGitSourceControlProvider& provider = GitSourceControlModule->GetProvider();
```
Same comment as RefreshLocks
This operation is quite quick, but we may as well hoist this to happen immediately after the safeguard - we're storing a raw pointer to GitSourceControlModule, it existing at the top of this function doesn't guarantee it exists here

Damon3000s replied:
Done.

Damon3000s committed:
Review feedback

- Hoist usages of GitSourceControlModule pointer so they happen immediately and not after other git operations have occurred.
